### PR TITLE
[mono][jit] Fix an assertion on x86+android.

### LINKED ISSUE #70302: .NET 7 Preview 6, AOT Compilation fails on Android
### Description

This started happening here: https://github.com/xamarin/xamarin-android/pull/7071

It appears that `System.Collections.Concurrent.dll` fails to be AOT-compiled:
```
/Users/runner/work/1/s/xamarin-android/bin/Release/dotnet/packs/Microsoft.Android.Sdk.Darwin/33.0.0-ci.darc-main-6f6e2198-9e92-4d2f-85f7-15e6c6aac8e9.38/targets/Microsoft.Android.Sdk.Aot.targets(78,5): Precompiling failed for /Users/runner/work/1/a/TestRelease/06-06_13.36.45/temp/DirectBootAwareAttribute/obj/Release/android-x86/linked/System.Collections.Concurrent.dll.
Mono Ahead of Time compiler - compiling assembly /Users/runner/work/1/a/TestRelease/06-06_13.36.45/temp/DirectBootAwareAttribute/obj/Release/android-x86/linked/System.Collections.Concurrent.dll
AOTID D4E0B35F-301C-B32D-6DC4-08A8DBC72CA4
Using profile data file '/Users/runner/work/1/s/xamarin-android/bin/Release/dotnet/packs/Microsoft.Android.Sdk.Darwin/33.0.0-ci.darc-main-6f6e2198-9e92-4d2f-85f7-15e6c6aac8e9.38/targets/dotnet.aotprofile'
Added 6 methods from profile.
* Assertion at /Users/runner/work/1/s/src/mono/mono/mini/method-to-ir.c:13068, condition `((sreg == -1) && (regtype == ' ')) || ((sreg != -1) && (regtype != ' '))' not met [/Users/runner/work/1/a/TestRelease/06-06_13.36.45/temp/DirectBootAwareAttribute/UnnamedProject.csproj]
```

This might cause any `Release` build to fail?

### Reproduction Steps

* `dotnet new android` using .NET SDK 7.0.100-preview.6.22306.4
* `dotnet build -c Release` (defaults to profiled AOT)

### Expected behavior

The `<MonoAOTCompiler/>` task succeeds.

### Actual behavior

The `<MonoAOTCompiler/>` task fails.

### Regression?

Yes, this was working in .NET SDK 7.0.100-preview.6.22277.6, which had a preview 5 runtime.

### Known Workarounds

n/a

### Configuration

.NET SDK 7.0.100-preview.6.22306.4

### Other information

Contents of an integration test project, `.binlog` inside: [Test Results - MSBuild One .NET - macOS-3 .zip](https://github.com/dotnet/runtime/files/8847536/Test.Results.-.MSBuild.One.NET.-.macOS-3.zip)

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/dotnet/runtime/issues/70302.

### CODE REVIEW

File: src/mono/mono/mini/mini.c [hunk 1]
```diff
@@ -3758,6 +3758,7 @@ mini_method_compile (MonoMethod *method, guint32 opts, JitFlags flags, int parts
 		cfg->init_method_rgctx_ins = NULL;
 		cfg->init_method_rgctx_ins_arg->opcode = OP_PCONST;
 		cfg->init_method_rgctx_ins_arg->inst_p0 = NULL;
+		MONO_INST_NULLIFY_SREGS (cfg->init_method_rgctx_ins_arg);
```
Comment (lambdageek) [inline src/mono/mono/mini/mini.c:3761]: is it surprising that init_method_rgctx_ins_arg has non-zero sregs? It seems like it should always be an OP_AOTCONST or an OP_PCONST

**Review by lambdageek — APPROVED** [review]

File: src/mono/mono/mini/mini.c (hunk 1 above)
Comment (lambdageek) [reply src/mono/mono/mini/mini.c:3761]: oh, I see. it's `OP_GOT_ENTRY` that has a `(dest)->inst_basereg = __got_loc->dreg;` which is `sreg1`

**Comment by SamMonoRT** [comment]
@fanyang-mono - any special lane we should look for to validate this builds fine - as in x86 Android, do we have any ? 
@vargaz - once the initial runs complete, we should do extra plaforms and wasm runs for completeness.

**Comment by vargaz** [comment]
This happens on android+x86+aot, not sure any lane is testing that. I tested manually.

**Pull request merged**